src/stream.c

```diff
@@ -1167,21 +1172,32 @@ uvc_error_t uvc_stream_get_frame(uvc_stream_handle_t *strmh,
       int err = pthread_cond_timedwait(&strmh->cb_cond, &strmh->cb_mutex, &ts);
 
       //TODO: How should we handle EINVAL?
+      if (err) {
+        pthread_mutex_unlock(&strmh->cb_mutex);
+      }
       switch(err){
       case EINVAL:
           *frame = NULL;
           return UVC_ERROR_OTHER;
       case ETIMEDOUT:
           *frame = NULL;
+          if (strmh->transfer_status == LIBUSB_TRANSFER_NO_DEVICE) {
+              UVC_DEBUG("no device - broken");
+              return UVC_ERROR_NO_DEVICE;
+          }
           return UVC_ERROR_TIMEOUT;
       }
     }
-    
-    if (strmh->last_polled_seq < strmh->hold_seq) {
+    if (strmh->transfer_status == LIBUSB_TRANSFER_NO_DEVICE) {
```

Comment:
Error conditions should go below success conditions; i.e. put this as the `else if` instead of the `if`.